src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Api.Enduser/Controllers/ConnectionController.cs

```diff
@@ -22,281 +23,209 @@ namespace Altinn.AccessManagement.Api.Enduser.Controllers;
-    /// <param name="from">The GUID identifying the party the authenticated user is acting for</param>
-    /// <param name="to">The GUID identifying the target party to which the assignment should be created.</param>
-    /// <param name="cancellationToken"><see cref="CancellationToken"/></param>
     [HttpGet]
+    [DbAudit(Claim = AltinnCoreClaimTypes.PartyUuid, System = AuditDefaults.EnduserApiStr)]
     [Authorize(Policy = AuthzConstants.POLICY_ACCESS_MANAGEMENT_ENDUSER_READ)]
-    [ServiceFilter(typeof(AuthorizePartyUuidClaimFilter))]
-    [ProducesResponseType<AssignmentExternal>(StatusCodes.Status200OK, MediaTypeNames.Application.Json)]
-    [ProducesResponseType<ProblemDetails>(StatusCodes.Status400BadRequest, MediaTypeNames.Application.Json)]
+    [ProducesResponseType<PaginatedResult<AssignmentExternal>>(StatusCodes.Status200OK, MediaTypeNames.Application.Json)]
+    [ProducesResponseType<AltinnProblemDetails>(StatusCodes.Status400BadRequest, MediaTypeNames.Application.Json)]
     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status403Forbidden)]
-    public async Task<IActionResult> GetConnection([FromQuery] Guid party, [FromQuery] Guid? from, [FromQuery] Guid? to, CancellationToken cancellationToken = default)
+    public async Task<IActionResult> GetConnections([FromQuery] ConnectionInput connection, [FromQuery, FromHeader] PagingInput paging, CancellationToken cancellationToken = default)
     {
-        if (!from.HasValue && !to.HasValue)
+        var userUuid = Audit.Value;
+        if (ValidationRules.EnduserGetConnection(userUuid.ChangedBy, connection.Party, connection.From, connection.To) is var problem && problem is { })
         {
-            return BadRequest();
+            return problem.ToActionResult();
         }
 
-        if (!(from.HasValue && from.Value == party) && !(to.HasValue && to.Value == party))
+        var result = await ConnectionService.Get(from: connection.From.ConvertToUuid(userUuid.ChangedBy), to: connection.To.ConvertToUuid(userUuid.ChangedBy), cancellationToken: cancellationToken);
+        if (result.IsProblem)
         {
-            // Party must match From or To
-            return BadRequest();
+            return result.Problem.ToActionResult();
         }
 
-        var partyUuid = Accessor.GetPartyUuid();
-        var audit = new ChangeRequestOptions()
-        {
-            ChangedBy = partyUuid,
-            ChangedBySystem = AuditDefaults.EnduserApi
-        };
-
-        if (from.HasValue && to.HasValue)
-        {
-            return Ok(await connectionService.Get(fromId: from.Value, toId: to.Value, facilitatorId: null, cancellationToken: cancellationToken));
-        }
+        return Ok(PaginatedResult.Create(result.Value, null));
+    }
 
-        if (from.HasValue)
+    /// <summary>
+    /// Get connections between the authenticated user's selected party and the specified target party.
+    /// </summary>
+    [HttpGet("{id}")]
```

Comment:
Ikke ett API vi har noen plan for å bruke i noen kontekst p.t.

Reply:
@Thuen